File: SECRET_PROJECT_WEB/src/types/Message/Message.ts
export type Message = {
  id: string;
  content?: string;
  senderId: string;
  reciverId: string;
  sentAt: string;
  file?: File;
  repliedMessage?: RepliedMessage;
  reactions?: MessageReaction[];
};

export type MessageReaction = {
  id: string;
  messageId: string;
  userId: string;
  emotion: string;
};

export type RepliedMessage = {
  repliedMessageId: string;
  senderName: string;
  content: string;
  file?: File;
};

export type File = {
  fileUrl?: string;
  fileType?: string;
  fileName?: string;
};

export type MessageDeleteRequest = {
  messageId: string;
  forAllUsers: boolean;
};

export type GetMessagesRequest = {
  firstUserId: string;
  secondUserId: string;
  page?: number;
};

File: SECRET_PROJECT_SERVER/Secret_Project_Backend/Secret_Project_Backend/Models/Message.cs
using System.ComponentModel.DataAnnotations;

namespace Secret_Project_Backend.Models
{
    public class Message
    {
        [Key]
        public Guid Id { get; set; }
        public string SenderId { get; set; }
        public virtual ApplicationUser Sender { get; set; }
        public string ReciverId { get; set; }
        public virtual ApplicationUser Reciver { get; set; }
        public DateTime SentAt { get; set; }
        public string? Content { get; set; }
        public string? FileUrl { get; set; }
        public string? FileType { get; set; }
    }
}
